packages/main/src/Badge.js

```diff
@@ -32,6 +32,22 @@ const metadata = {
 			type: String,
 			defaultValue: "1",
 		},
+
+		/**
+		 * Defines if the badge has an icon.
+		 * @private
+		 */
+		__hasIcon: {
+			type: Boolean,
+		},
+
+		/**
+		 * Defines if the badge has only an icon (and no text).
+		 * @private
+		 */
+		__hasIconOnly: {
```

Comment:
same here 
Moreover isIconOnly or just iconOnly is more exact

Reply:
Done